tests/smtp_test.go

```diff
@@ -68,6 +69,94 @@ func TestCheckRequireTLS(tt *testing.T) {
 	conn.ExpectPattern("221 *")
 }
 
+func TestProxyProtocolTrustedSource(tt *testing.T) {
+	tt.Parallel()
+	t := tests.NewT(tt)
+	t.DNS(map[string]mockdns.Zone{
+		"one.maddy.test.": {
+			TXT: []string{"v=spf1 ip4:192.168.99.1 -all"},
+		},
+	})
+	t.Port("smtp")
+	t.Config(`
+		smtp tcp://127.0.0.1:{env:TEST_PORT_smtp} {
+			hostname mx.maddy.test
+			tls off
+
+			proxy_protocol {
+				trust ` + tests.DefaultSourceIP.String() + ` ::1/128
+				tls off
+			}
+
+			defer_sender_reject no
+
+			check {
+				spf {
+					enforce_early yes
+					fail_action reject
+				}
+			}
+
+			deliver_to dummy
+		}
+	`)
+	t.Run(1)
+	defer t.Close()
+
+	conn := t.Conn("smtp")
+	defer conn.Close()
+	conn.Writeln(fmt.Sprintf("PROXY TCP4 192.168.99.1 %s 12345 %d", tests.DefaultSourceIP.String(), t.Port("smtp")))
```

Comment:
Hardcoded IP address may be not exactly a best approach. Is this fine as it is, or should I add another constant (`SecondarySourceIP`)?

Reply:
This seems confusing.  Testing code binds on DefaultSourceIP before the connection is made and connects to 127.0.0.1.

Shouldn't it be `PROXY TCP4 <DefaultSourceIP> 127.0.0.1 12345 smtp`?

Reply:
Probably using `127.0.0.17` would be better to avoid any traffic if test misbehaves and actually tries to contact 192.168.99.1.